File: wlr/OGUR/OGUR/Dungeons/PointPoint.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OGUR.Dungeons
{
    class PointPoint
    {
        public readonly int X;
        public readonly int Y;
        private readonly bool _isHorizontal;

        public PointPoint(int x, int y, bool destroyHorizontal = false)
        {
            X = x;
            Y = y;
            _isHorizontal = destroyHorizontal;
        }
        public bool isHorizontal()
        {
            return _isHorizontal;
        }
    }
}

File: proj/src/aigilas/dungeons/PointPoint.java
package aigilas.dungeons;

class PointPoint {
    public int X;
    public int Y;
    private boolean _isHorizontal;

    public PointPoint(int x, int y, boolean destroyHorizontal) {
        X = x;
        Y = y;
        _isHorizontal = destroyHorizontal;
    }

    public PointPoint(int x, int y) {
        this(x, y, false);
    }

    public boolean isHorizontal() {
        return _isHorizontal;
    }
}
